packages/hdwallet-ledger-webusb/src/adapter.ts

```diff
@@ -45,56 +51,60 @@ export class WebUSBLedgerAdapter {
   private async handleDisconnectWebUSBLedger(e: USBConnectionEvent): Promise<void> {
     if (e.device.vendorId !== VENDOR_ID) return;
 
+    this.keyring.emit(
+      [e.device.manufacturerName ?? "", e.device.productName ?? "", core.Events.DISCONNECT],
+      e.device.serialNumber
+    );
+
     const ts = Date.now();
     this.currentEventTimestamp = ts;
 
     // timeout gives time to detect if it is an app navigation based disconnect/connect event
     // discard disconnect event if it is not the most recent event received
-    setTimeout(async () => {
-      if (ts !== this.currentEventTimestamp) return;
-
-      try {
-        if (e.device.serialNumber) await this.keyring.remove(e.device.serialNumber);
-      } catch (error) {
-        console.error(error);
-      } finally {
-        this.keyring.emit(
-          [e.device.manufacturerName ?? "", e.device.productName ?? "", core.Events.DISCONNECT],
-          e.device.serialNumber
-        );
-      }
-    }, APP_NAVIGATION_DELAY);
+    // TODO(gomes): maybe uncomment me maybe not
```

Comment:
Most likely not, as we've seen, trying to detect Ledger connected is pretty much impossible, and our UX should make it so we only connect at key times, meaning actual "connection" is really irrelevant cc @kaladinlight

Reply:
removing